packages/panna-sdk/src/core/onramp/onramp-money/session-status.ts

```diff
@@ -0,0 +1,124 @@
+ * @param params.client - The Panna client instance used for authentication
+ * @param params.authToken - Optional JWT token for authentication
+ * @returns Promise resolving to the session status with all transaction details
+ * @throws Error if the session ID is invalid, client is unauthorized, or network request fails
+ *
+ * @example
+ * ```ts
+ * import { getSessionStatus } from 'panna-sdk/core';
+ *
+ * // Check the status of an onramp.money session
+ * const status = await getSessionStatus({
+ *   sessionId: "session-123",
+ *   client: pannaClient,
+ *   authToken: "your-jwt-token" // optional
+ * });
+ *
+ * // Handle different status types
+ * switch (status.status) {
+ *   case 'created':
+ *     console.log('Session created, waiting for payment');
+ *     break;
+ *
+ *   case 'pending':
+ *     console.log('Payment received, processing onramp...');
+ *     break;
+ *
+ *   case 'completed':
+ *     console.log('Onramp completed successfully!');
+ *     console.log('Transaction hash:', status.transaction_hash);
+ *     break;
+ *
+ *   case 'failed':
+ *     console.error('Onramp failed:', status.error_message);
+ *     break;
+ *
+ *   case 'cancelled':
+ *     console.log('Session was cancelled');
+ *     break;
+ *
+ *   case 'expired':
+ *     console.log('Session expired');
+ *     break;
+ * }
+ * ```
+ */
+export async function getSessionStatus(
+  params: GetSessionStatusParams
+): Promise<SessionStatusResult> {
+  const { sessionId, client, authToken } = params;
+
+  if (!sessionId) {
+    throw new Error('Session ID is required');
+  }
+
+  if (!client) {
+    throw new Error('Panna client is required');
+  }
+
+  // Mock mode for testing
+  if (IS_MOCK_MODE) {
+    // Simulate API delay
+    await new Promise((resolve) => setTimeout(resolve, 500));
+
+    // Return mock data based on session ID pattern for testing
+    if (sessionId.includes('created')) {
+      return mockSessionStatusCreated;
+    } else if (sessionId.includes('pending')) {
+      return mockSessionStatusPending;
+    } else {
+      return mockSessionStatusCompleted;
+    }
+  }
+
+  try {
+    const url = `${PANNA_API_URL}/api/v1/onramp/session/${sessionId}`;
+
+    const headers: HeadersInit = {
+      'Content-Type': 'application/json'
+    };
+
+    // Add Authorization header if authToken is provided
+    if (authToken) {
+      headers.Authorization = `Bearer ${authToken}`;
+    }
+
+    const response = await fetch(url, {
+      method: 'GET',
+      headers
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to get session status: ${response.status} ${response.statusText}`
+      );
+    }
+
+    const result = (await response.json()) as SessionStatusResult;
+    return result;
+  } catch (error) {
+    throw new Error(
+      `Failed to get onramp.money session status for ${sessionId}: ${
+        error instanceof Error ? error.message : 'Unknown error'
+      }`
+    );
+  }
```

Comment:
The catch block wraps errors, potentially losing important details from the original error (like network errors, CORS issues, etc.).

To preserve the original error as the cause:
```
catch (error) {
  const message = `Failed to get onramp.money session status for ${sessionId}`;
  throw new Error(message, { cause: error });
}
```